fix: stop sending mod acks on receipt when auto ack extensions are disabled

hannahrogers-google opened the pull request:
If auto Ack Extensions are disabled (maxAckExtensionPeriod is 0), we should not send a ModAck request when a message is first received.

Fixes #141 ☕️

elharo reviewed (commented):
I'm confused about "mod" versus "mock" in the description. This doesn't look like a mock. Can you update the description to clarify?

Also, if this isn't a mock, then a test for this bug would be very helpful. Thanks.

hannahrogers-google commented:
> I'm confused about "mod" versus "mock" in the description. This doesn't look like a mock. Can you update the description to clarify?

Sorry about that, typo - I have fixed the description to be mod instead of mock

> Also, if this isn't a mock, then a test for this bug would be very helpful. Thanks.

I have added a test as well.

kamalaboulhosn commented:
Are we still going to send the receipt ack? If we don't, then the deadline won't be set properly and regardless of the setting, the deadline will be 10 minutes.
